lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1076,14 +1076,26 @@ static void optimizeSliceOfSplat(Function *F) {
 }
 
 /// Eliminate ReshapeNode when the input is already the correct shape.
+/// Also eliminate ReshapeNode when the input is SplatNode.
```

Comment:
I'd suggest moving the `  /// Eliminate ReshapeNode when the input is already the correct shape. ` comment into the body of the function, right before the corresponding `if`. And removing the `/// Also eliminate ReshapeNode when the input is SplatNode` here.

We may add more reshape optimizations in the future. But it doesn't make a lot of sense to list them all in the function's comment.

IMHO, the function's comment should simply say something like:
`/// Optimize reshape nodes.`

Reply:
Adopted.